server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -405,7 +407,8 @@ public BytesReference cacheKey(CheckedBiConsumer<ShardSearchRequest, StreamOutpu
                 differentiator.accept(this, out);
             }
             // copy it over since we don't want to share the thread-local bytes in #scratch
-            return out.copyBytes();
+            BytesReference bytesReference = out.copyBytes();
+            return new BytesArray(sha256().digest(bytesReference.array()));
```

Comment:
There is no need to `copyBytes` if all we need is the hash.

Reply:
Pushed a commit without copying bytes now